packages/blade/src/components/Form/Selector/SelectorInput.web.tsx

```diff
@@ -57,20 +60,30 @@ const StyledInput = styled.input<HoverProps>(({ theme, isChecked, isDisabled, ha
   },
 }));
 
-const SelectorInput = ({
-  inputProps,
-  isChecked,
-  isDisabled,
-  hasError,
-}: HoverProps & { inputProps: any }): React.ReactElement => {
+const _SelectorInput: React.ForwardRefRenderFunction<
+  BladeElementRef,
+  HoverProps & { inputProps: any }
+> = ({ inputProps, isChecked, isDisabled, hasError }, ref) => {
+  const inputRef = useBladeInnerRef(ref);
+
   return (
     <StyledInput
       isChecked={isChecked}
       isDisabled={isDisabled}
       hasError={hasError}
       {...inputProps}
+      // merging both refs because inputProps.ref needs to have access to indeterminate state
+      // to be able to set the mixed value via setMixed() function
+      // TODO: replace with a generic `mergeRefs()` util if we do this in other places
+      ref={(value) => {
+        inputProps.ref.current = value;
```

Comment:
> where is this inputprops coming from? 

`inputProps` is coming form the `useCheckboxHook` which sets/handles all the core logic for the checkbox DOM element. 


<img width="446" alt="image" src="https://user-images.githubusercontent.com/35374649/212839084-6373286d-2423-4b1d-9f54-d512189a9692.png">

> why are there 2 refs now?

We need the `inputProps.ref` because we need to set the indeterminate state of the input element which can only be set via the DOM element attribute. `inputRef.indeterminate = true;`

There are two refs:

1. the inputProps.ref which we need to handle the indeterminate state
2. the one which gets exposed to the public (with useBladeInnerRef)

Reply:
ah ok makes sense